internal/juju/offers.go

```diff
@@ -150,10 +144,15 @@ func (c offersClient) CreateOffer(input *CreateOfferInput) (*CreateOfferResponse
 		return nil, errs
 	}
 
+	modelOwner, modelName, err := c.SharedClient.ModelOwnerAndName(input.ModelUUID)
```

Comment:
GetModelOwnerFromCache
GetModelNameFromCache

Would be clearer, because it could look like an API call